### PR TITLE
media-video/obs-studio: add amf support

### PR DESCRIPTION
**Pull request by MrDuartePT**
This PR have the intention to add streamfx plugin support for obs-studio like the aur that have obs-studio-streamfx pkgbuild

Prove of version 29.1.3 working:
image

The 9999 ebuild not seems to compile but it not streamfx related issue as shown by the build log (is the same with our without the flag)

build.log
```
 * Package:    media-video/obs-studio-9999:0
 * Repository: gentoo
 * Maintainer: chiitoo@gentoo.org
 * USE:        abi_x86_64 alsa amd64 browser elibc_glibc kernel_linux lua_single_target_luajit nvenc pipewire pulseaudio python python_single_target_python3_11 ssl truetype v4l vlc wayland websocket
 * FEATURES:   network-sandbox preserve-libs sandbox userpriv usersandbox
 * Using python3.11 to build
>>> Unpacking source...
>>> Unpacking cef_binary_5060_linux64.tar.bz2 to /var/tmp/portage/media-video/obs-studio-9999/work
 * Repository id: obsproject_obs-studio.git
 * To override fetched repository properties, use:
 *   EGIT_OVERRIDE_REPO_OBSPROJECT_OBS_STUDIO
 *   EGIT_OVERRIDE_BRANCH_OBSPROJECT_OBS_STUDIO
 *   EGIT_OVERRIDE_COMMIT_OBSPROJECT_OBS_STUDIO
 *   EGIT_OVERRIDE_COMMIT_DATE_OBSPROJECT_OBS_STUDIO
 *
 * Fetching https://github.com/obsproject/obs-studio.git ...
git fetch https://github.com/obsproject/obs-studio.git +HEAD:refs/git-r3/HEAD
git symbolic-ref refs/git-r3/media-video/obs-studio/0/__main__ refs/git-r3/HEAD
 * Skipping submodule plugins/win-dshow/libdshowcapture
 * Using submodule plugins/obs-browser
 * Skipping submodule plugins/obs-outputs/ftl-sdk
 * Using submodule plugins/obs-websocket
 * Repository id: obsproject_obs-browser.git
 * To override fetched repository properties, use:
 *   EGIT_OVERRIDE_REPO_OBSPROJECT_OBS_BROWSER
 *   EGIT_OVERRIDE_BRANCH_OBSPROJECT_OBS_BROWSER
 *   EGIT_OVERRIDE_COMMIT_OBSPROJECT_OBS_BROWSER
 *   EGIT_OVERRIDE_COMMIT_DATE_OBSPROJECT_OBS_BROWSER
 *
 * Fetching https://github.com/obsproject/obs-browser.git ...
git fetch https://github.com/obsproject/obs-browser.git --prune +refs/heads/*:refs/heads/* +refs/tags/*:refs/tags/* +refs/notes/*:refs/notes/* +refs/pull/*/head:refs/pull/* +HEAD:refs/git-r3/HEAD
git update-ref --no-deref refs/git-r3/media-video/obs-studio/0/plugins_obs-browser/__main__ 2ba72dbdfed60acd38fa1ac7a52cbc6617ae8fd1
 * Repository id: obsproject_obs-websocket.git
 * To override fetched repository properties, use:
 *   EGIT_OVERRIDE_REPO_OBSPROJECT_OBS_WEBSOCKET
 *   EGIT_OVERRIDE_BRANCH_OBSPROJECT_OBS_WEBSOCKET
 *   EGIT_OVERRIDE_COMMIT_OBSPROJECT_OBS_WEBSOCKET
 *   EGIT_OVERRIDE_COMMIT_DATE_OBSPROJECT_OBS_WEBSOCKET
 *
 * Fetching https://github.com/obsproject/obs-websocket.git ...
git fetch https://github.com/obsproject/obs-websocket.git --prune +refs/heads/*:refs/heads/* +refs/tags/*:refs/tags/* +refs/notes/*:refs/notes/* +refs/pull/*/head:refs/pull/* +HEAD:refs/git-r3/HEAD
git update-ref --no-deref refs/git-r3/media-video/obs-studio/0/plugins_obs-websocket/__main__ 4ff109b62bc221192943541010d055be9ae5dbba
 * Checking out https://github.com/obsproject/obs-studio.git to /var/tmp/portage/media-video/obs-studio-9999/work/obs-studio-9999 ...
git checkout --quiet refs/git-r3/HEAD
GIT update -->
   repository:               https://github.com/obsproject/obs-studio.git
   at the commit:            6b5404899a98ced94a695bcb429db846d90b51c0
 * Skipping submodule plugins/win-dshow/libdshowcapture
 * Using submodule plugins/obs-browser
 * Skipping submodule plugins/obs-outputs/ftl-sdk
 * Using submodule plugins/obs-websocket
 * Checking out https://github.com/obsproject/obs-browser.git to /var/tmp/portage/media-video/obs-studio-9999/work/obs-studio-9999/plugins/obs-browser ...
git checkout --quiet 2ba72dbdfed60acd38fa1ac7a52cbc6617ae8fd1
GIT update -->
   repository:               https://github.com/obsproject/obs-browser.git
   at the commit:            2ba72dbdfed60acd38fa1ac7a52cbc6617ae8fd1
 * Checking out https://github.com/obsproject/obs-websocket.git to /var/tmp/portage/media-video/obs-studio-9999/work/obs-studio-9999/plugins/obs-websocket ...
git checkout --quiet 4ff109b62bc221192943541010d055be9ae5dbba
GIT update -->
   repository:               https://github.com/obsproject/obs-websocket.git
   at the commit:            4ff109b62bc221192943541010d055be9ae5dbba
>>> Source unpacked in /var/tmp/portage/media-video/obs-studio-9999/work
>>> Preparing source in /var/tmp/portage/media-video/obs-studio-9999/work/obs-studio-9999 ...
 * Source directory (CMAKE_USE_DIR): "/var/tmp/portage/media-video/obs-studio-9999/work/obs-studio-9999"
 * Build directory  (BUILD_DIR):     "/var/tmp/portage/media-video/obs-studio-9999/work/obs-studio-9999_build"
>>> Source prepared.
>>> Configuring source in /var/tmp/portage/media-video/obs-studio-9999/work/obs-studio-9999 ...
 * Source directory (CMAKE_USE_DIR): "/var/tmp/portage/media-video/obs-studio-9999/work/obs-studio-9999"
 * Build directory  (BUILD_DIR):     "/var/tmp/portage/media-video/obs-studio-9999/work/obs-studio-9999_build"
cmake -C /var/tmp/portage/media-video/obs-studio-9999/work/obs-studio-9999_build/gentoo_common_config.cmake -G Ninja -DCMAKE_INSTALL_PREFIX=/usr -DCEF_ROOT_DIR=../cef_binary_5060_linux64 -DCALM_DEPRECATION=ON -DCCACHE_SUPPORT=OFF -DENABLE_ALSA=yes -DENABLE_AJA=OFF -DENABLE_BROWSER=yes -DENABLE_DECKLINK=no -DENABLE_FREETYPE=yes -DENABLE_JACK=no -DENABLE_LIBFDK=no -DENABLE_NEW_MPEGTS_OUTPUT=no -DENABLE_PIPEWIRE=yes -DENABLE_PULSEAUDIO=yes -DENABLE_QSV11=no -DENABLE_RNNOISE=ON -DENABLE_RTMPS=ON -DENABLE_SPEEXDSP=no -DENABLE_V4L2=yes -DENABLE_VLC=yes -DENABLE_VST=ON -DENABLE_WAYLAND=yes -DENABLE_WEBRTC=OFF -DENABLE_WEBSOCKET=yes -DOBS_MULTIARCH_SUFFIX=64 -DUNIX_STRUCTURE=1 -DENABLE_SCRIPTING_LUA=no -DENABLE_SCRIPTING_PYTHON=yes -DENABLE_SCRIPTING=ON -DENABLE_WHATSNEW=ON -DCMAKE_BUILD_TYPE=RelWithDebInfo -DCMAKE_TOOLCHAIN_FILE=/var/tmp/portage/media-video/obs-studio-9999/work/obs-studio-9999_build/gentoo_toolchain.cmake /var/tmp/portage/media-video/obs-studio-9999/work/obs-studio-9999
loading initial cache file /var/tmp/portage/media-video/obs-studio-9999/work/obs-studio-9999_build/gentoo_common_config.cmake
-- OBS:  Application Version: 30.0.0-rc1-64-g6b5404899-modified - Build Number: 1
-- The C compiler identification is Clang 17.0.4
-- The CXX compiler identification is Clang 17.0.4
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/lib/llvm/17/bin/clang - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/lib/llvm/17/bin/clang++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found OpenGL: /usr/lib64/libOpenGL.so
-- Found FFmpeg: /usr/lib64/../lib64/libavcodec.so (found version "#define LIBAVCODEC_VERSION_MINOR   3;#define LIBAVCODEC_VERSION_MICRO 100.3.100") found components: avcodec avdevice avutil avformat
-- Found CURL: /usr/lib64/libcurl.so (found version "8.4.0")
-- OBS:  DISABLED   Luajit support
-- Found Python: /var/tmp/portage/media-video/obs-studio-9999/temp/python3.11/bin/python3 (found version "3.11.5") found components: Interpreter Development Development.Module Development.Embed
-- Found SWIG: /usr/bin/swig (found suitable version "4.1.1", minimum required is "3")
-- OBS:  ENABLED    obspython
-- OBS:  obs-scripting -> Python 3.11.5 found.
-- Found X11: /usr/include
-- Looking for XOpenDisplay in /usr/lib64/libX11.so;/usr/lib64/libXext.so
-- Looking for XOpenDisplay in /usr/lib64/libX11.so;/usr/lib64/libXext.so - found
-- Looking for gethostbyname
-- Looking for gethostbyname - found
-- Looking for connect
-- Looking for connect - found
-- Looking for remove
-- Looking for remove - found
-- Looking for shmat
-- Looking for shmat - found
-- Looking for IceConnectionNumber in ICE
-- Looking for IceConnectionNumber in ICE - found
-- Found PkgConfig: /usr/bin/x86_64-pc-linux-gnu-pkg-config (found version "1.8.1")
-- XCB[XCB]: Found component XCB
-- Found XCB: /usr/lib64/libxcb.so  found components: XCB
-- Found X11_XCB: /usr/lib64/libX11-xcb.so
-- Found OpenGL: /usr/lib64/libOpenGL.so  found components: EGL
-- Found Wayland: /usr/lib64/libwayland-client.so;/usr/lib64/libwayland-server.so;/usr/lib64/libwayland-egl.so;/usr/lib64/libwayland-cursor.so
-- Found Jansson: /usr/lib64/../lib64/libjansson.so (found suitable version "2.14", minimum required is "2.5")
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE
-- Found FFmpeg: /usr/lib64/libavformat.so (found version "#define LIBAVFORMAT_VERSION_MINOR   3;#define LIBAVFORMAT_VERSION_MICRO 100.3.100") found components: avformat avutil swscale swresample avcodec
-- Found ZLIB: /usr/lib64/libz.so (found version "1.2.13")
-- Found LibUUID: /usr/lib64/libuuid.so
-- OBS:  -> PulseAudio found - audio monitoring enabled
-- Checking for modules 'gio-2.0;gio-unix-2.0'
--   Found gio-2.0, version 2.76.4
--   Found gio-unix-2.0, version 2.76.4
-- Found Gio: /usr/lib64/libgio-2.0.so
-- Found Wayland: /usr/lib64/libwayland-client.so;/usr/lib64/libwayland-server.so;/usr/lib64/libwayland-egl.so;/usr/lib64/libwayland-cursor.so  found components: Client
-- Found Xkbcommon: /usr/lib64/libxkbcommon.so
-- Performing Test COMPILER_HAS_HIDDEN_VISIBILITY
-- Performing Test COMPILER_HAS_HIDDEN_VISIBILITY - Success
-- Performing Test COMPILER_HAS_HIDDEN_INLINE_VISIBILITY
-- Performing Test COMPILER_HAS_HIDDEN_INLINE_VISIBILITY - Success
-- Performing Test COMPILER_HAS_DEPRECATED_ATTR
-- Performing Test COMPILER_HAS_DEPRECATED_ATTR - Success
-- XCB[XCB]: Found component XCB
-- XCB[XFIXES]: Found component XFIXES
-- XCB[RANDR]: Found component RANDR
-- XCB[SHM]: Found component SHM
-- XCB[XINERAMA]: Found component XINERAMA
-- XCB[COMPOSITE]: Found component COMPOSITE
-- Found XCB: /usr/lib64/libxcb.so;/usr/lib64/libxcb-xfixes.so;/usr/lib64/libxcb-randr.so;/usr/lib64/libxcb-shm.so;/usr/lib64/libxcb-xinerama.so;/usr/lib64/libxcb-composite.so  found components: XCB XFIXES RANDR SHM XINERAMA COMPOSITE
-- OBS:  ENABLED    linux-capture
-- OBS:  ENABLED    linux-pulseaudio
-- Found Libv4l2: /usr/lib64/libv4l2.so
-- Found FFmpeg: /usr/lib64/libavcodec.so (found version "#define LIBAVCODEC_VERSION_MINOR   3;#define LIBAVCODEC_VERSION_MICRO 100.3.100") found components: avcodec avutil avformat
-- Found Udev: /usr/lib64/libudev.so
-- OBS:  ENABLED    linux-v4l2
-- OBS:  DISABLED   linux-jack
-- Found ALSA: /usr/lib64/libasound.so (found version "1.2.9")
-- OBS:  ENABLED    linux-alsa
-- Found PipeWire: /usr/lib64/libpipewire-0.3.so (found suitable version "0.3.80", minimum required is "0.3.33")
-- OBS:  ENABLED    linux-pipewire
-- OBS:  DISABLED   decklink
-- Found LibVLC: /usr/include/vlc
-- OBS:  ENABLED    vlc-video
-- OBS:  DISABLED   sndio
-- Performing Test HAVE_STDATOMIC
-- Performing Test HAVE_STDATOMIC - Success
-- Found WrapAtomic: TRUE
-- Found OpenGL: /usr/lib64/libOpenGL.so
-- Found WrapOpenGL: TRUE
-- Found XKB: /usr/lib64/libxkbcommon.so (found suitable version "1.5.0", minimum required is "0.5.0")
-- Found WrapVulkanHeaders: /usr/include
-- OBS:    -        obs-vst uses bundled VST headers
-- OBS:  ENABLED    obs-vst
-- OBS:  DISABLED   obs-qsv11
-- Found Chromium Embedded Framework: /var/tmp/portage/media-video/obs-studio-9999/work/cef_binary_5060_linux64/Release/libcef.so;
-- Found CEF: /var/tmp/portage/media-video/obs-studio-9999/work/cef_binary_5060_linux64/Release/libcef.so
-- Found nlohmann_json: /usr/share/cmake/nlohmann_json/nlohmann_jsonConfig.cmake (found version "3.11.2")
-- OBS:  ENABLED    obs-browser-page
-- OBS:  ENABLED    obs-browser
-- Found nlohmann_json: /usr/share/cmake/nlohmann_json/nlohmann_jsonConfig.cmake (found suitable version "3.11.2", minimum required is "3")
CMake Warning (dev) at cmake/Modules/Findqrcodegencpp.cmake:108 (message):
  Failed to find qrcodegencpp version.
Call Stack (most recent call first):
  plugins/obs-websocket/cmake/legacy.cmake:22 (find_package)
  cmake/Modules/ObsHelpers.cmake:470 (include)
  plugins/obs-websocket/CMakeLists.txt:3 (legacy_check)
This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Error at /usr/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
  Could NOT find qrcodegencpp (missing: qrcodegencpp_LIBRARY
  qrcodegencpp_INCLUDE_DIR) (found version "0.0.0")

      Reason given by package: Ensure that qrcodegencpp is installed on the system.

Call Stack (most recent call first):
  /usr/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:600 (_FPHSA_FAILURE_MESSAGE)
  cmake/Modules/Findqrcodegencpp.cmake:135 (find_package_handle_standard_args)
  plugins/obs-websocket/cmake/legacy.cmake:22 (find_package)
  cmake/Modules/ObsHelpers.cmake:470 (include)
  plugins/obs-websocket/CMakeLists.txt:3 (legacy_check)


-- Configuring incomplete, errors occurred!
 * ERROR: media-video/obs-studio-9999::gentoo failed (configure phase):
 *   cmake failed
 *
 * Call stack:
 *     ebuild.sh, line  136:  Called src_configure
 *   environment, line 4346:  Called cmake_src_configure
 *   environment, line 2111:  Called die
 * The specific snippet of code:
 *       "${CMAKE_BINARY}" "${cmakeargs[@]}" "${CMAKE_USE_DIR}" || die "cmake failed";
 *
 * If you need support, post the output of `emerge --info '=media-video/obs-studio-9999::gentoo'`,
 * the complete build log and the output of `emerge -pqv '=media-video/obs-studio-9999::gentoo'`.
 * The complete build log is located at '/var/tmp/portage/media-video/obs-studio-9999/temp/build.log'.
 * The ebuild environment file is located at '/var/tmp/portage/media-video/obs-studio-9999/temp/environment'.
 * Working directory: '/var/tmp/portage/media-video/obs-studio-9999/work/obs-studio-9999_build'
 * S: '/var/tmp/portage/media-video/obs-studio-9999/work/obs-studio-9999'
```

### CODE REVIEW

**Comment by Chiitoo** [comment]
Thanks!

I'll try to look into this soon.

And yeah, 'qrcodegencpp', which is required by USE="websocket", is not packaged yet.  It's basically a mix of two different upstreams, and not something I'm looking forward to maintain...

**Comment by MrDuartePT** [comment]
Yha i also notice that it in guru.

I also gonna add the amf patch if you dont mind i get the patch form obs-studio-amf of from aur but i think is this PR where they get it: https://github.com/obsproject/obs-studio/pull/7206

**Comment by MrDuartePT** [comment]
I gonna push the amf fetaure sortly

**Comment by MrDuartePT** [comment]
It working just need to add the obs-studio dektop file to launch using the amd_pro_icd64.json

**Comment by MrDuartePT** [comment]
well after add the amf depedency i getting this error in pkgcheck:
```
  NonsolvableDepsInDev: version 29.1.3-r3: nonsolvable depset(depend) keyword(~amd64) dev profile (default/linux/amd64/17.0/musl) (5 total): solutions: [ media-video/amdgpu-pro-amf, media-video/ffmpeg[amf(+)] ]
  NonsolvableDepsInDev: version 29.1.3-r3: nonsolvable depset(rdepend) keyword(~amd64) dev profile (default/linux/amd64/17.0/musl) (5 total): solutions: [ media-video/amdgpu-pro-amf, media-video/ffmpeg[amf(+)] ]
  NonsolvableDepsInStable: version 29.1.3-r3: nonsolvable depset(depend) keyword(~arm64) stable profile (default/linux/arm64/17.0) (37 total): solutions: [ media-video/amdgpu-pro-amf, media-video/ffmpeg[amf(+)] ]
  NonsolvableDepsInStable: version 29.1.3-r3: nonsolvable depset(rdepend) keyword(~arm64) stable profile (default/linux/arm64/17.0) (37 total): solutions: [ media-video/amdgpu-pro-amf, media-video/ffmpeg[amf(+)] ]
```

**Comment by Chiitoo** [comment]
Take your time.  : ]

I will need sleep, so not getting into this tonight, as there will be work tomorrow.

But yeah, 'media-video/amdgpu-pro-amf' is only keyworded for 'amd64' at this time, so the USE-flag will need to be masked for the other arches that 'media-video/obs-studio' has been keyworded for.

**Comment by MrDuartePT** [comment]
> Take your time. : ]
> 
> I will need sleep, so not getting into this tonight, as there will be work tomorrow.
> 
> But yeah, 'media-video/amdgpu-pro-amf' is only keyworded for 'amd64' at this time, so the USE-flag will need to be masked for the other arches that 'media-video/obs-studio' has been keyworded for.

Well for my part is done at least the ci didn’t failed :]
I also gonna sleep, when you review this give me a ping if need any changes.

File: media-video/obs-studio/obs-studio-9999.ebuild [hunk 1]
```diff
@@ -216,13 +243,13 @@ src_configure() {
 			-DENABLE_SCRIPTING=ON
 		)
 	else
-		mycmakeargs+=( -DENABLE_SCRIPTING=OFF )
+		mycmakeargs+=(-DENABLE_SCRIPTING=OFF)
```
Comment (scardracs) [inline media-video/obs-studio/obs-studio-9999.ebuild:246]: isn't the += supposed to maintain the spaces?

Comment (thesamesam) [reply media-video/obs-studio/obs-studio-9999.ebuild:246]: It's just style, the expansion happens later.

Comment (MrDuartePT) [reply media-video/obs-studio/obs-studio-9999.ebuild:246]: but if you what i can revert to the old style

Comment (scardracs) [reply media-video/obs-studio/obs-studio-9999.ebuild:246]: > but if you what i can revert to the old style

Nono don’t worry. IIRC some years ago was an error but if Sam says it’s ok I’m more than ok with it

**Comment by MrDuartePT** [comment]
I added the desktop because of bug I found on ffmpeg ebuild that need to be fix: https://github.com/gentoo/gentoo/pull/33851

TLDR: ffmpeg compile with amf set AMDGPU Pro Vulkan icd loader as the default system wide

**Comment by Chiitoo** [comment]
Sorry I haven't had the chance to look into this with thought, but I'll try to do so soon.

I may push 30.0.0 before it though, if I manage to decide what to do with USE="websocket" (might mask it temporarily).

If you want to start basing this on that instead, you can find a not-entirely-finished PR of it here:

https://github.com/gentoo/gentoo/pull/33910

Either way works for me though.  I may be able to do this before I decide that, but in any case, I hope I manage to do one or the other during this week!

Thank you!

**Comment by MrDuartePT** [comment]
No problem. 
If you prefer I update this PR after you release 0.30.0

File: media-video/obs-studio/metadata.xml [hunk 2]
```diff
@@ -23,6 +23,8 @@
     <flag name="ssl">Build with TLS/SSL support for RTMPS connections.</flag>
     <flag name="vlc">Build with VLC media source support.</flag>
     <flag name="websocket">Build with WebSocket API support.</flag>
+    <flag name="streamfx">Build with StreamFx plugin support (Experimental).</flag>
+    <flag name="amf">Build with AMD AMF patch (Experimental).</flag>
```
Comment (Chiitoo) [inline media-video/obs-studio/metadata.xml:27]: These could be listed in an alphabetical order.

File: media-video/obs-studio/obs-studio-9999.ebuild [hunk 3]
```diff
@@ -3,16 +3,17 @@
 
 EAPI=8
 
-CMAKE_REMOVE_MODULES_LIST=( FindFreetype )
-LUA_COMPAT=( luajit )
-PYTHON_COMPAT=( python3_{9..12} )
+CMAKE_REMOVE_MODULES_LIST=(FindFreetype)
+LUA_COMPAT=(luajit)
+PYTHON_COMPAT=(python3_{9..12})
```
Comment (Chiitoo) [inline media-video/obs-studio/obs-studio-9999.ebuild:8]: Editor automatically making style changes perhaps?

File: media-video/obs-studio/obs-studio-9999.ebuild [hunk 4]
```diff
@@ -3,16 +3,17 @@
...
 
-inherit cmake lua-single optfeature python-single-r1 xdg
+inherit desktop cmake lua-single optfeature python-single-r1 xdg
```
Comment (Chiitoo) [inline media-video/obs-studio/obs-studio-9999.ebuild:10]: Does desktop need to be before cmake here?

If I remember right, there are cases when that is true, but usually these are also listed in an alpabetical order.

**Comment by Chiitoo** [comment]
Didn't notice anything after a very quick look, aside from the possibly editor auto-changes and alphabetical listing things, but will hopefully get a better look-see and testing in too soon!

I pushed 30.0.0, so indeed you might want to re-work this on top of that.

Thanks!

File: media-video/obs-studio/metadata.xml (hunk 2 above)
Comment (MrDuartePT) [reply media-video/obs-studio/metadata.xml:27]: Ok I will make the change

File: media-video/obs-studio/obs-studio-9999.ebuild (hunk 3 above)
Comment (MrDuartePT) [reply media-video/obs-studio/obs-studio-9999.ebuild:8]: Yes was the editor changing the style

File: media-video/obs-studio/obs-studio-9999.ebuild (hunk 4 above)
Comment (MrDuartePT) [reply media-video/obs-studio/obs-studio-9999.ebuild:10]: No need to be before I gonna change to alphabetical order

**Comment by MrDuartePT** [comment]
@Chiitoo is updated.
Please review it to see if everything is alright

Thanks in advance

**Comment by MrDuartePT** [comment]
> Thanks for looking into this @MrDuartePT,
> 
> My original plan was to actually do exactly this, but then I kinda gave up when [obsproject/obs-studio#7206](https://github.com/obsproject/obs-studio/pull/7206) got closed (not merged) and that StreamFX explicitly deprecated AMF support since it got added to obs on Windows.
> 
> I suppose it's nice to at least have native AMF support for the OBS versions where it works.

Well the patch was extracted from the PR (github provide the diff file if you add .diff to the end of url) and I just notice the AUR package for obs-amf was using the latest version with the patch, just follow the same ideia here and also in Nobara they have a obs package that support amf as well

I was also reading the PR and It seems they don't merge it because it dosen't work on flatpack, because amd don't provide they driver on flatpack like nvidia thus (well this is not fault of the PR is more of problem with amd, but well I don't manage the Obs project)

That is only one missing feature on igpu compare to windows, H.265 support (I have a friend that I have the same cpu on his laptop and can use H.265 on windows), for some reason the obs-amf-test always report false, even if I disable the nvidia dgpu, for amd dgpu H.265 works.

obs-amf-test output:
```
```

**Comment by habibm1361** [comment]
The best laptop of the day for those of you who are interested in artificial intelligence...
XNote - ChatGPT-Powered Smart Notebook
https://xnote-chatgpt-powered-smart.kckb.st/076220a1

**Comment by AdelKS** [comment]
Is the AMF support patch on obs is the one that is on-top of the texture encoding PR ?

**Comment by MrDuartePT** [comment]
> Is the AMF support patch on obs is the one that is on-top of the texture encoding PR ? This branch: [obsproject/obs-studio#8312 (comment)](https://github.com/obsproject/obs-studio/pull/8312#issuecomment-1492127806)

I dont fix it is

**Comment by MrDuartePT** [comment]
@Chiitoo I just solve something nothing major.
I notice one thing the 0.30.0 ebuild when i try to install Streamfx dosent get clone if failt to resolve the github url.
It a bit strange since src_unpack dosent block network access.

**Comment by AdelKS** [comment]
> but I will wait for the https://github.com/obsproject/obs-studio/pull/8312 to get merge

Yeah sounds good !

**Comment by AdelKS** [comment]
Is there a reason to have `streamfx` part of the `obs-studio` ebuild ?

File: media-video/obs-studio/obs-studio-30.0.0-r1.ebuild [hunk 5]
```diff
@@ -0,0 +1,294 @@
...
+
+	sed -i '/-Werror$/d' "${WORKDIR}"/${P}/cmake/Modules/CompilerConfig.cmake || die
+
+	if use streamfx; then
+		echo 'add_subdirectory(streamfx)' &>>"${WORKDIR}"/${P}/UI/frontend-plugins/CMakeLists.txt || die
+	fi
+
+	if use amf; then
+		cd "${S}" || die
+		eapply "${FILESDIR}/${PN}-amf-${PV}.patch" || die
```
Comment (AdelKS) [inline media-video/obs-studio/obs-studio-30.0.0-r1.ebuild:201]: the patch doesn't exist for obs 30

**Comment by MrDuartePT** [comment]
> Is there a reason to have `streamfx` part of the `obs-studio` ebuild ?

Well since streamfx doesn't include binary anymore, it a bit more involved to install the plugin. And since I like to use the plugin I think would be a good addition to be already include.

But if gonna be a bit troublesome I can just scratch the ideia or make separate PR until I find a better way to add streamfx

File: media-video/obs-studio/obs-studio-30.0.0-r1.ebuild (hunk 5 above)
Comment (MrDuartePT) [reply media-video/obs-studio/obs-studio-30.0.0-r1.ebuild:201]: Thanks for the catch I forget to change the patch name. I will update it tomorrow

**Comment by AdelKS** [comment]
I tried to write a quick ebuild for obs-streamfx as a plugin and indeed it's not plug-and-play as e.g. [obs-multi-rtmp](https://github.com/gentoo/guru/blob/master/media-video/obs-multi-rtmp/obs-multi-rtmp-9999.ebuild): the project has a very complex structure, with quite the number of (embedded) dependencies.

By looking at its [BUILDING.md](https://github.com/Xaymar/obs-StreamFX/blob/root/BUILDING.md), the supported way to build streamFX is to embed its code within obs' code :/ But there's what he calls "CI-style" to build it outside which is considered unsupported. 

Given the complexity of the streamFX module, I don't think obs' ebuild should be made even more complex with this extra module.

**Comment by AdelKS** [comment]
I just tested the version 29.1.3 with the `amf` useflag: when obs-studio is run in "amf" mode (with the ICD override to amdgpu-pro-vulkan), VAAPI encoding is still possible (does it go through mesa ?). So I think we can simply override the .desktop entry with the ICD override when the `amf` useflag is enabled. No need for two desktop entries, since the "amf" entry has everything the non-amf one offers ? :thinking:

**Comment by MrDuartePT** [comment]
> I tried to write a quick ebuild for obs-streamfx as a plugin and indeed it's not plug-and-play as e.g. [obs-multi-rtmp](https://github.com/gentoo/guru/blob/master/media-video/obs-multi-rtmp/obs-multi-rtmp-9999.ebuild): the project has a very complex structure, with quite the number of (embedded) dependencies.

It not very plug and play I just try but just give up because of the problems I was getting

> 
> By looking at its [BUILDING.md](https://github.com/Xaymar/obs-StreamFX/blob/root/BUILDING.md), the supported way to build streamFX is to embed its code within obs' code :/ But there's what he calls "CI-style" to build it outside which is considered unsupported.
>

I can try to ask the streamFX developer how they do the old binaries or see the old docs. Or just try the CIstyle and try to do a ebuild from there

> Given the complexity of the streamFX module, I don't think obs' ebuild should be made even more complex with this extra module.

You probably right is the better approach to do for now. I will tomorrow remove the streamFX from the PR.
@Chiitoo your what me to do a new PR just with amf or just change this one?

**Comment by MrDuartePT** [comment]
> I just tested the version 29.1.3 with the `amf` useflag: when obs-studio is run in "amf" mode (with the ICD override to amdgpu-pro-vulkan), VAAPI encoding is still possible (does it go through mesa ?). So I think we can simply override the .desktop entry with the ICD override when the `amf` useflag is enabled. No need for two desktop entries, since the "amf" entry has everything the non-amf one offers ? 🤔

Also the NVIDIA encoder also work when the amf desktop entry. I also will try vaapi tomorrow (never verify it was working) but thanks for the feedback. If it works as intended I will only use one desktop file.

**Comment by AdelKS** [comment]
> The only to not require the override is setting the ICD system wide (not a good ideia😅).

here's another idea: modify the patch so obs calls ffmpeg with the ICD override :thinking:

**Comment by MrDuartePT** [comment]
> > The only to not require the override is setting the ICD system wide (not a good ideia😅).
> 
> here's another idea: modify the patch so obs calls ffmpeg with the ICD override 🤔

It a good ideia. I can try to do that in later time when I change patch for the texture encoding (if the PR get merge). I need to get familiar with the obs code and my C++ knowledge is kinda rusty😅.

**Comment by MrDuartePT** [comment]
I gonna do all the change and then take the draft out.

**Comment by MrDuartePT** [comment]
I gonna get the patch from this (repo)[https://github.com/nowrep/obs-studio/tree/amf-linux-tex] and add to amf patch. I gonna just remove the texture patches

**Comment by MrDuartePT** [comment]
Well @AdelKS I can't find a way to make load ffmpeg with the ICD override but I gonna change the Desktop file to use `AMD_VULKAN_ICD=amdvlk-pro` instead since is the recommend way by AMD.
For what Im seeing nvidia just check if the nvidia module is loaded and set the card to be use with NVENC. Since amdgpu-pro only install ICD it not be possible to make that kind of check.

**Comment by MrDuartePT** [comment]
@Chiitoo @AdelKS I think is ready.
@AdelKS I will also will update the other [PR](https://github.com/gentoo/gentoo/pull/33851) and improve the streamfx ebuild and push it to guru

**Comment by MrDuartePT** [comment]
Well send a email to the created of amf patch and he said to use vaapi instead since the problems with vaapi where fix. https://github.com/obsproject/obs-studio/pull/9179

I think I can just scratch this PR and only focuses on the obs-streamfx. Or just merge with the support for 29.1.3. Also the mantainer of aur package close the package because of the same problem with 30.0.0 and the vaapi fix

@Chiitoo you want to merge the path for 29.1.3?

**Comment by AdelKS** [comment]
Great job @MrDuartePT ! One good thing done ! You did not have to keep me as an author for the commit x)

File: media-video/obs-studio/files/com.obsproject.Studio.amf.desktop [hunk 6]
```diff
@@ -0,0 +1,97 @@
+[Desktop Entry]
+Version=1.0
+Name=OBS Studio (AMF)
+GenericName=Streaming/Recording Software
+Comment=Free and Open Source Streaming/Recording Software
+Exec=env AMD_VULKAN_ICD=amdvlk-pro obs
```
Comment (AdelKS) [inline media-video/obs-studio/files/com.obsproject.Studio.amf.desktop:6]: As we discovered `AMD_VULKAN_ICD` doesn't work, so here we either need to use `vk_pro` (thus we need to wait for the other PR to get merged) or explicitly do the VK_DRIVER_FILES override.

I think that Intel also does encoding through Vulkan, do you have an Intel iGPU to test ? Otherwise I think we probably should simply drop this PR :/

**Comment by Chiitoo** [comment]
> I think I can just scratch this PR and only focuses on the obs-streamfx. Or just merge with the support for 29.1.3. Also the mantainer of aur package close the package because of the same problem with 30.0.0 and the vaapi fix
> 
> @Chiitoo you want to merge the path for 29.1.3?

Probably not if/when this is going to be its separate package, which I think does look like the better path forward.

Thanks again!

**Comment by MrDuartePT** [comment]
> > I think I can just scratch this PR and only focuses on the obs-streamfx. Or just merge with the support for 29.1.3. Also the mantainer of aur package close the package because of the same problem with 30.0.0 and the vaapi fix
> > @Chiitoo you want to merge the path for 29.1.3?
> 
> Probably not if/when this is going to be its separate package, which I think does look like the better path forward.
> 
> Thanks again!

Sorry I think I misspelled I was only referring to the amf patch. The last version it compatible is 29.1.3 since in 30.0.0 obs fixed the vaapi bugs.

Streamfx plugin in now on guru I gonna try to see when I have time to do also gstreamer plugin.

**Comment by AdelKS** [comment]
I suppose this PR needs to be closed then

**Comment by MrDuartePT** [comment]
Yes is better to close this since 29.1.3 is a old version anyway and it more easy to just use vappi, is not requires any additional drivers, just work with mesa.
